tests/functional/xtriggers/04-suite_state.t

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Test that deprecation warnings are printed appropriately for the suite_state
+# xtrigger.
```

Comment:
Does this functional test provide anything more than the unit tests below?

Reply:
It's icky to test the deprecation message logging from within unit tests because it happens at import time, so did this as a functional test. https://github.com/cylc/cylc-flow/pull/5864#discussion_r1424290455